toolkit/tools/specreader/specreader.go

```diff
@@ -82,13 +84,55 @@ func main() {
 	toolchainRPMs, err := schedulerutils.ReadReservedFilesList(*toolchainManifest)
 	logger.PanicOnError(err, "Unable to read toolchain manifest file '%s': %s", *toolchainManifest, err)
 
-	err = parseSPECsWrapper(*buildDir, *specsDir, *rpmsDir, *srpmsDir, *existingToolchainRpmDir, *distTag, *output, *workerTar, toolchainRPMs, *workers, *runCheck)
+	// A parse list may be provided, if so only parse this subset.
+	// If none is provided, parse all specs.
+	parseList, err := parseParseListFile(*parseListFile)
 	logger.PanicOnError(err)
+
+	// Convert specsDir to an absolute path
+	specsAbsDir, err := filepath.Abs(*specsDir)
+	logger.PanicOnError(err, "Unable to get absolute path for specs directory '%s': %s", *specsDir, err)
+
+	err = parseSPECsWrapper(*buildDir, specsAbsDir, *rpmsDir, *srpmsDir, *existingToolchainRpmDir, *distTag, *output, *workerTar, parseList, toolchainRPMs, *workers, *runCheck)
+	logger.PanicOnError(err)
+}
+
+// parsePackListFile will parse a list of packages to pack if one is specified.
+// Duplicate list entries in the file will be removed.
+func parseParseListFile(parseListFile string) (parseList map[string]bool, err error) {
```

Comment:
Might be worth mentioning that the `bool` is a dummy value and that you are treating the `map` as a set.

Reply:
For better or worse we use this pattern all over the codebase, I tend to name them 'sets' when I use them. I'll update this function to follow that pattern, although I'm going to leave the original version of this function alone for now.